src/System.Linq/tests/AverageTests.cs

```diff
@@ -44,6 +38,25 @@ public void EmptyNullableFloatSource()
         }
 
         [Fact]
+        public void NullNFloatSource()
+        {
+            Assert.Throws<ArgumentNullException>("source", () => default(IEnumerable<float?>).Average());
```

Comment:
@jaredpar, in our CI system, this (and a bunch of these other tests) is producing the warning as error:
```
AverageTests.cs(43,66): error CS1720: Warning as Error: Expression will always cause a System.NullReferenceException because the default value of 'System.Collections.Generic.IEnumerable<float?>' is null [d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Linq\tests\System.Linq.Tests.csproj]
```
Is that just a bug in the old C# compiler?  These are extension methods, so this warning is wrong.